egs/wham/ConvTasNet/eval.py

```diff
@@ -1,14 +1,16 @@
+#!/usr/bin/env python3
```

Comment:
Can you elaborate on the advantage of that?

Reply:
Like we discussed, this is called `shebang` and it will allows the user to automatically use the python from his/her virtual environment.

Reply:
The advantage of the other technique is that, without changing anything the user's env is used. But if the user wants, he can point to another env, so that it'll always be the same. 
This can also be done by activating the env by hand in run.sh. 

So, not a big advantage for the current way. I'd be willing to change that. 
@jonashaag, point of view? Mitigations?

Reply:
I don’t understand what shebang has to do with virtualenv. If you use `python` as a relative path, no matter if it’s used as part of shebang or written explicitly in a shell scripts, it will always use the virtualenv Python. The only case where the system Python will be used is if you use absolute paths to the binary but I don’t think we are doing that anywhere (if so we definitely should change it to relative)

Reply:
Had a quick look at how the eval script is invoked in the current version—the shebang isn’t used at all so this change is a no-op.

i actually think the old version with the global Python interpreter path is more flexible since you can tell it to use a different Python interpreter to run the scripts than is currently used by any virtualenv installations (if you have any). It also allows you to run the scripts without activating the virtualenv in the first place if you want that.

Reply:
yes so I revised the shebang behavihor and found the same thing. I removed it already. Thanks a lot for such fruitful advice @jonashaag 

"The only case where the system Python will be used is if you use absolute paths to the binary but I don’t think we are doing that anywhere (if so we definitely should change it to relative)" -> yes that's what I will be a bit concerned for people who are hacking scripts to make for their own user cases. I met it before: It happened when I used python from brew (well, brew is a terrible tool to be played with in terms of linking and compatibility control though.....)

Reply:
>  actually think the old version with the global Python interpreter path is more flexible since you can tell it to use a different Python interpreter to run the scripts than is currently used by any virtualenv installations (if you have any). It also allows you to run the scripts without activating the virtualenv in the first place if you want that.

I agree with this, and this was the original intent. I wonder if it's really necessary though..